analyzers/src/SonarAnalyzer.CFG/LiveVariableAnalysis/RoslynLiveVariableAnalysis.cs

```diff
@@ -174,6 +175,20 @@ private void BuildBranchesNestedCatchRethrow(BasicBlock block)
         }
     }
 
+    private void BuildBranchesRethrowFinally(BasicBlock block)
```

Comment:
I think we can merge this method with the method I added for the similar case (catch -> outercatch connection if there's a rethrow) like so:

```csharp
     private void BuildBranchesRethrow(BasicBlock block)
    {
        var reachableHandlerRegions = block.EnclosingRegion(ControlFlowRegionKind.TryAndCatch).NestedRegion(ControlFlowRegionKind.Try).ReachableHandlers();
        var reachableCatchAndFinallyBlocks = reachableHandlerRegions.Where(x => x.FirstBlockOrdinal > block.Ordinal).SelectMany(x => x.Blocks(Cfg));
        foreach (var catchBlock in reachableCatchAndFinallyBlocks.Where(x => x.EnclosingRegion.Kind is ControlFlowRegionKind.Catch))
        {
            AddBranch(block, catchBlock);
        }
        if (reachableCatchAndFinallyBlocks.FirstOrDefault() is { } branch)
        {
            AddBranch(block, branch);
        }
    }   
```

WDYT?